root/static/scripts/edit/URLCleanup.js

```diff
@@ -3210,6 +3209,13 @@ const CLEANUPS = {
       return url.replace(/^(?:https?:\/\/)?(?:[^/]+\.)?weibo\.com\/(u\/)?([^\/?#]+)(?:.*)$/, 'https://www.weibo.com/$1$2');
     },
   },
+  'whosampled': {
+    match: [new RegExp('^(https?://)?(www\\.)?whosampled\\.com', 'i')],
+    type: LINK_TYPES.otherdatabases,
+    clean: function (url) {
+      return url.replace(/^(?:https?:\/\/)?(?:www\.)?whosampled\.com\/(.*)+$/, 'https://www.whosampled.com/$1');
+    },
+  },
```

Comment:
It would be nice to add validation to prevent `?` and `#` signs to be added instead of their percent encoding form.
For example, see https://musicbrainz.org/url/9b6219cf-9893-4485-a8dd-3d2f9579e970

Reply:
How would you envision this validation working? Just reject any URLs containing those? Or clean them up to their encoded variants? Or something else?

Reply:
_Just reject any URLs containing those_ would be nice at least. Cleaning them up can be difficult because it would require to distinguish between misencoded `?` in the slug from `?` which is used for the query part. Let’s not over-engineer it.